### PR TITLE
[REFACTOR] domain/subnet dropdowns refactor

### PR DESCRIPTION
**Pull request by ohadlevy**
This code removes dupliation across host/group edit for primary
nic and additional interfaces.

This is the first step towards removing the old html based js responses
for chained select input

### CODE REVIEW

File: app/views/hostgroups/_form.html.erb [hunk 1]
```diff
@@ -40,7 +40,17 @@
     </div>
 
     <div class="tab-pane" id="network">
-      <%= render 'common/domain_subnet', :item => @hostgroup %>
+      <%= select_f f, :domain_id, accessible_domains, :id, :to_label,
+                   { :include_blank => accessible_domains.any? ? true : _('No domains')},
+                   { :disabled => accessible_domains.empty? ? true : false,
```
Comment (domcleal) [inline app/views/hostgroups/_form.html.erb:45]: nitpick, I don't think `? true : false` is needed here and in the other places

**Comment by domcleal** [comment]
:+1: looks good to me.  @abenari, would you like to review too?

**Comment by ohadlevy** [comment]
rebased

**Comment by domcleal** [comment]
[test]

**Comment by domcleal** [comment]
The new host form no longer loads, could you look please? :)

<pre>
  1) Error:
HostTest#test_0002_create new page:
ActionView::MissingTemplate: Missing partial common/domain with {:locale=>[:en], :formats=>[:html], :handlers=>[:erb, :builder, :coffee, :rabl]}. Searched in:
  * "/home/dcleal/code/foreman/foreman/app/views"
</pre>

**Comment by ohadlevy** [comment]
thanks... looks like a bad merge

**Comment by domcleal** [comment]
ping @ohadlevy

**Comment by dLobatog** [comment]
Rebase again please.

"This is the first step towards removing the old html based js responses
for chained select input"

What follows? Compute resources js, hostgroups js..?

**Comment by ohadlevy** [comment]
we have duplicate code between interfaces (selecting domain/subnet ) and
primary nic (same selections for domain/subnet), next imho is the
arch/os/media/ptable etc.

**Comment by dLobatog** [comment]
@ohadlevy Can you rebase? This seems nearly ready to merge after fixing the `true : false` thing @domcleal pointed out. If you don't plan to work on this soon please close it, let me know and I'll pick it up and resubmit.

**Comment by dLobatog** [comment]
@ohadlevy Ping? Rebasing is still possible and required fixes are minor, no need to lose this contribution (and the time you spent writing it :smile: )

**Comment by ohadlevy** [comment]
Not sure if the latest nic work does not invalidates this PR...

**Comment by dLobatog** [comment]
Closing this after the latest changes on NIC rendering this PR unusable, @ohadlevy feel free to reopen, sorry!

**Comment by ohadlevy** [comment]
:+1 thanks!

**Pull request closed without merging**